### PR TITLE
docs: Add linking statement for the terminologies below

### PR DESCRIPTION
**Pull request by datsabk**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Good catch! The lack of a bridge statement was really awkward.

File: docs/core_concepts.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Core Concepts
 
-Let's assume you're familiar with core Git, Docker, Kubernetes, Continuous Delivery, and GitOps concepts.
+Let's assume you're familiar with core Git, Docker, Kubernetes, Continuous Delivery, and GitOps concepts. Let us understand some of the terminologies pertaining to ArgoCD.
```
[the remark this answers is not in the document]
Comment (datsabk) [reply docs/core_concepts.md:3]: I think below statement would sound better. 

`Let's familiarize ourselves with the concepts that are specific to Argo CD`

[the remark this answers is not in the document]
Comment (crenshaw-dev) [reply docs/core_concepts.md:3]: Yes, that's much better than mine!

[the remark this answers is not in the document]
Comment (datsabk) [reply docs/core_concepts.md]: Done! Pushed the change with commit sign off as requested

**Comment by crenshaw-dev** [comment]
Closing in favor of https://github.com/argoproj/argo-cd/pull/9036

**Pull request closed without merging**